### ISSUE TITLE
Specify dim reduction and classification algorithms using a string

### ISSUE THREAD

**Issue by olgabot** [YeoLab/flotilla#209]
It would be great if the user could specify a dimensionality reduction algorithm without having to write their own class.

Currently, this is how the user would specify tSNE:

``` python
from sklearn.manifold import tSNE

class DataFrametSNE(DataFrameReducerBase, tSNE):
    pass

study.plot_dimensionality_reduction(reducer=DataFrametSNE)
```

This is cumbersome and not encouraging to the iterative process as it requires the user to step out of the hypothesis generation mindset.

Ideally, the `plot_dimensionality_reduction` code would evaluate the reducer under the hood, and create a class on-the-fly for this reducer. We would need to double-check that this reducer actually works for this data (e.g. for NMF you need only non-negative data), but we can just throw an error saying this reducer isn't compatible with the data.

Here's what I envision:

``` python
study.plot_dimensionality_reduction(reducer='tSNE')
```
